metacat-common/src/main/java/com/netflix/metacat/common/QualifiedName.java

```diff
@@ -291,6 +298,47 @@ public static QualifiedName ofTable(
         return new QualifiedName(catalogName, databaseName, tableName, null, null);
     }
 
+    private static QualifiedName createCaseSensitiveQualifiedNameIfNecessary(
+        @NonNull final String catalogName,
```

Comment:
I don't have the full context, but I would create a separate function with the non-standardlized qualifiedname as the single input, then it returns the standard qualifiedName as one object. We can also create a new class as the standizedQualified name (which exends the current one) to handle the format and related operations.

Reply:
Agreed, but the expectation is that this is a one-off requirement and I wanted to avoid having to refactor all the QualifiedName constructors in the Controller/Service code to call a new API or a new QualifiedName type. This way, the change is contained inside the QualifiedName itself and the Connector makes a decision on whether to use the case-sensitive version or not.